Cu-f33z0z move sytax to core

nomisRev opened the pull request:
This PR moves the Arrow Syntax module to Arrow Core Data, in a subsequent PR both modules will be moved to Arrow Core.

franciscodr commented:
Task linked: [CU-f33z0z Move arrow-syntax into arrow-core](https://app.clickup.com/t/f33z0z)

rachelcarmena commented:
Hi @nomisRev :raised_hands: 

This directory (outside a module) appeared again:

![](https://user-images.githubusercontent.com/22792183/110315418-81763200-8009-11eb-82a3-b02018b5aa45.png)

Thanks!!

aballano reviewed (approved):
just a nit with tests' packages

rachelcarmena commented on arrow-stack/build.gradle:
```diff
@@ -26,7 +26,6 @@ dependencies {
         api "io.arrow-kt:arrow-meta:$VERSION_NAME"
         api "io.arrow-kt:arrow-optics:$VERSION_NAME"
         api "io.arrow-kt:arrow-optics-test:$VERSION_NAME"
-        api "io.arrow-kt:arrow-syntax:$VERSION_NAME"
```
:clap:

rachelcarmena commented on arrow-libs/core/arrow-core-data/src/main/kotlin/arrow/syntax/function/composition.kt:
```diff
@@ -1,4 +1,4 @@
-package arrow.syntax.function
+package arrow.core
```
Maybe it could be moved to that directory as well :raised_hands:

rachelcarmena reviewed (approved):
:rocket: Thank you @nomisRev !!

nomisRev commented:
@aballano & @rachelcarmena I addressed both your comments. I moved the files to `arrow.core` and I moved the tests there as well.

I'm not sure if I correctly got rid of the directory though @rachelcarmena. I'm having some issues with it since it was not showing up in my diff locally as removed.

rachelcarmena reviewed (approved):
:clap:  It was removed :raised_hands: Maybe it doesn't appear because Git doesn't add empty directories.Thanks @nomisRev !!
